components/infobox/wikis/mobilelegends/infobox_league

```diff
@@ -0,0 +1,339 @@
+local Tier = require('Module:Tier')
+local PageLink = require('Module:Page')
+local Json = require('Module:Json')
+local Class = require('Module:Class')
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Title = require('Module:Infobox/Widget/Title')
+local Center = require('Module:Infobox/Widget/Center')
+
+local CustomLeague = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local _args
+
+local _ABBR_USD = '<abbr title="United States Dollar">USD</abbr>'
+local _TODAY = os.date('%Y-%m-%d', os.time())
+
+function CustomLeague.run(frame)
+	local league = League(frame)
+	_args = league.args
+
+	league.addToLpdb = CustomLeague.addToLpdb
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
+	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
+
+	return league:createInfobox(frame)
+end
+
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'gamesettings' then
+		return {
+			Cell{name = 'Patch', content = {
+					CustomLeague._getPatchVersion()
+				}},
+			}
+	elseif id == 'prizepool' then
+		return {
+			Cell{
+				name = 'Prize pool',
+				content = {CustomLeague:_createPrizepool()},
+			},
+		}
+	elseif id == 'liquipediatier' then
+		return {
+			Cell{
+				name = 'Liquipedia tier',
+				content = {CustomLeague:_createTierDisplay()},
+				classes = {_args['moonton-sponsored'] == 'true' and 'valvepremier-highlighted' or ''},
+			},
+		}
+	elseif id == 'customcontent' then
+		if _args.player_number then
+			table.insert(widgets, Title{name = 'Players'})
+			table.insert(widgets, Cell{name = 'Number of players', content = {_args.player_number}})
+		end
+
+		--teams section
+		if _args.team_number or (not String.isEmpty(_args.team1)) then
+			Variables.varDefine('is_team_tournament', 1)
+			table.insert(widgets, Title{name = 'Teams'})
+		end
+		table.insert(widgets, Cell{name = 'Number of teams', content = {_args.team_number}})
+		if not String.isEmpty(_args.team1) then
+			local teams = CustomLeague:_makeBasedListFromArgs('team')
+			table.insert(widgets, Center{content = teams})
+		end
+
+		--maps
+		if not String.isEmpty(_args.map1) then
+			table.insert(widgets, Title{name = 'Maps'})
+			table.insert(widgets, Center{content = CustomLeague:_makeMapList()})
+		end
+	end
+	return widgets
+end
+
+function League:addToLpdb(lpdbData, args)
+	lpdbData.participantsnumber = args.player_number or args.team_number
+
+	return lpdbData
+end
+
+function League:defineCustomPageVariables()
+	if _args.player_number then
+		Variables.varDefine('tournament_mode', 'solo')
+	end
+	Variables.varDefine('tournament_patch', _args.patch)
+	Variables.varDefine('tournament_endpatch', _args.epatch)
+
+	Variables.varDefine('tournament_publishertier', _args['moonton-sponsored'])
+
+	--Legacy Vars:
+	Variables.varDefine('tournament_edate', Variables.varDefault('tournament_enddate'))
```

Comment:
Is this variable needed?

Reply:
> Is this variable needed?

I removed in 2nd but will temporarily add back in 3rd due to the issue I found

It appears that when I removed this and pushed live, In the team/player results table it will not recognize the end date, and always treat that tournament enddate to be UNIX (1970 01 01) adding back these two line cause the date to work again